### PR TITLE
Fix upstream rocm xla

### PR DESCRIPTION
**Pull request by rahulbatra85**
1) ROCM registration for the custom ops for GPU assert 
2) Update a ROCm specific check in  gpu_atomic_test

### CODE REVIEW

**Review by cheshire — CHANGES_REQUESTED** [review]

File: tensorflow/compiler/xla/service/gpu/runtime_intrinsics.cc [hunk 1]
```diff
@@ -72,7 +72,12 @@ static void AssertionCustomCall(void* stream_handle, void** buffers,
   }
 }
 
+#if GOOGLE_CUDA
 XLA_REGISTER_CUSTOM_CALL_TARGET_WITH_SYM(kXlaGpuAssertCustomCallTag,
                                          AssertionCustomCall, "CUDA");
+#else
+XLA_REGISTER_CUSTOM_CALL_TARGET_WITH_SYM(kXlaGpuAssertCustomCallTag,
```
[the remark this answers is not in the document]
Comment (rahulbatra85) [reply tensorflow/compiler/xla/service/gpu/runtime_intrinsics.cc:79]: Added a symbol, but had to merge this with another PR https://github.com/tensorflow/tensorflow/pull/56781 as both this PR and that has dependency on the new symbol

New PR is here https://github.com/tensorflow/tensorflow/pull/56793(merges 56781 and 56779)

[the remark this answers is not in the document]
Comment (rahulbatra85) [reply tensorflow/compiler/xla/service/gpu/runtime_intrinsics.cc:79]: Will close this

File: tensorflow/compiler/xla/service/gpu/tests/gpu_atomic_test.cc [hunk 2]
```diff
@@ -104,7 +104,7 @@ TEST_F(GpuAtomicTest, TestAddAtomicF32) {
 )";
 
   CompileAndVerifyIr(hlo_string, is_built_with_rocm_ ? R"(
-CHECK: atomicrmw fadd float addrspace{{.*}}, float {{.*}} seq_cst, align 4
+CHECK: atomicrmw fadd ptr addrspace(1) %[[ADDR:.*]], float %[[VALUE:.*]] syncscope("agent") seq_cst
```
[the remark this answers is not in the document]
Comment (rahulbatra85) [reply tensorflow/compiler/xla/service/gpu/tests/gpu_atomic_test.cc:107]: This is a ROCm specific check. Also, it built fine on Nvidia platform

**Comment by rahulbatra85** [comment]
Duplicated here https://github.com/tensorflow/tensorflow/pull/56793

**Pull request closed without merging**